From: Peter Krempa <pkrempa@redhat.com>
Date: Mon, 16 Mar 2015 16:52:44 +0100
Subject: [PATCH] qemu: block-commit: Mark disk in block jobs only on
 successful command

Patch 51f9f03a4ca50b070c0fbfb29748d49f583e15e1 introduces a regression
where if a blockCommit operation fails the disk is still marked as being
part of a block job but can't be unmarked later.

(cherry picked from commit ee744b5b387b5123ee40683c52ab40783ffc3020)
---
 src/qemu/qemu_driver.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/qemu/qemu_driver.c b/src/qemu/qemu_driver.c
index 3ec057b..4817e06 100644
--- a/src/qemu/qemu_driver.c
+++ b/src/qemu/qemu_driver.c
@@ -16781,7 +16781,8 @@ qemuDomainBlockCommit(virDomainPtr dom,
         goto endjob;
     }
 
-    disk->blockjob = true;
+    if (ret == 0)
+        disk->blockjob = true;
 
     if (mirror) {
         if (ret == 0) {
